### DISCUSSION TITLE
Flash not operational during reference image creation

### DISCUSSION THREAD

**Discussion by SGXander** [jomjol/AI-on-the-edge-device#3508] [General]
### The Problem

after alignment step 1, step 2 appears to disable the flash and use the rear LED on the board. This behaviour continues until the ESP is reset, where it works for alignment again but then breaks on step 2 repeatably

### Version

16.0.0-RC4

### Logfile

```shell
not at stage where menu available
```

### Expected Behavior

clicking create new reference button should produce the same image as alignment

### Screenshots

![image](https://github.com/user-attachments/assets/73a72e3b-8091-47fa-b33e-19523177809d)

### Additional Context

the ESP must be reset to restore flash functionality during alignment but the issue is repeatable

**Comment by mesistostefre** [comment]
Confirmed. Looks like the flash works just once and is disabled til next reset. Found no workaround. I'll roll back ...

**Comment by caco3** [comment]
Might be the same issue as on https://github.com/jomjol/AI-on-the-edge-device/issues/3338

**Comment by gotaede** [comment]
I can reproduce the error with firmware v15.7.0 and webinterface 16.0.0-RC4

**Comment by Kopetan4egX** [comment]
Same problem. Can't set reference image properly

**Comment by Kopetan4egX** [comment]
> I'll roll back ...

Which version was you flash working fine before? I see above my previous comment that problem was already present in previous release version 15.7.0

**Comment by SybexX** [comment]
@SGXander If you want you can try this firmware (16.0.0-RC4 + fix): https://github.com/SybexX/AI-on-the-edge-device/actions/runs/11874812323/artifacts/2197416682

**Comment by SGXander** [comment]
sorry for not updating, I rolled back to 15.7.0 too and it was ok there so I stuck with that. I also have an external flash too now and it's been fine. I'll look to update this weekend and see if the issue returns.

**Comment by SybexX** [comment]
There was no change in the LED implementation/Control between versions 15.4.0, 15.7.0 and 16.0.0-RC4.
If you had the version 15.7.0 html files on the SD and only updated the firmware.bin to version 16.0.0, could potentially experience the problem described.

**Comment by SGXander** [comment]
It's the other way round for me, I started with 16.0.0-RC4 as it was the top release and then realised it was RC so rebuilt the ESP32-CAM with 15.7.0

**Comment by SybexX** [comment]
@SGXander Did you have Auto-Exposure Control and Auto-Exposure Control 2 ("night mode") activated in both versions?

**Comment by SGXander** [comment]
can't say for certain before/on 16.0.0-RC4 but on my running version 15.7.0 it looks like this (believed default as I've not touched it)
![image](https://github.com/user-attachments/assets/57d24ec2-1020-4a64-8b2f-d23992a2be9b)

**Comment by doncorso** [comment]
Hello,
this sounds similar to my problem.
Preview on step 2 after first installation looks fine, but when I click "Create New Reference / Change Camera Settings" there is no flash ad because of this, I have only a dark picture as SGXander showed above.

Any ideas and/or updates ?

Steve Bee

**Comment by SybexX** [comment]
@doncorso please try this firmware: https://github.com/SybexX/AI-on-the-edge-device/actions/runs/11875592284/artifacts/2197579082

It would also be helpful if you described the exact procedure (what did you do and what files did you use, previous version or new installation............) as there are several ways to install and update AI-on-the-edge-device.

**Comment by jmealo** [comment]
I have this issue on a brand new installation.
I formatted the card as fat and copied over the remote_setup from the latest GitHub release.

**Comment by SybexX** [comment]
Is the LED really not turning on or are you just assuming it because the picture is dark?
There is no firmware included in remote-setup, only in manual-setup and update

remote-setup:

![remote](https://github.com/user-attachments/assets/704b0e01-3452-4347-b31f-034f3fc694da)

manual-setup:

![manual](https://github.com/user-attachments/assets/ce0efe2a-23ac-42bd-87e2-63edc12a0f5c)

update:

![update](https://github.com/user-attachments/assets/c645b22f-08c5-43a0-bb08-a1f783a36384)

**Comment by jmealo** [comment]
@SybexX: The flash turns off entirely after the preview. It never comes back on. If I power cycle it, it turns on briefly during the boot cycle, and only during the very first step. It doesn't work during the take reference image step.

**Comment by SybexX** [comment]
@jmealo Unfortunately, I can't tell from your statement which firmware you are using and how you flashed it onto the ESP32.

You can go through setup mode without making any changes or set "SetupMode = false" in sd-card/config/config.ini to bypass setup and check if this issue occurs generally or only in setup mode.

If you used the webinstaller and the files from remote-setup version 16.0.0-RC4, then the LED will not work in step 2 (Create New Reference).
- The webinstaller installs firmware version 15.7.0
- Firmware 15.7.0 + html files from 16.0.0 = The LED will not work in step 2.
- Firmware 16.0.0 + html files from 15.7.0 = The LED should work in step 2.

![Unbenannt](https://github.com/user-attachments/assets/02b98e5c-3b69-4e8d-b141-0dcfedec40ce)

**Comment by Kopetan4egX** [comment]
@SybexX i downloaded AI-on-the-edge-device__manual-setup__v15.7.0.zip, extracted files from sd-card folder to SD card, configured Wi-fi settings in wlan.ini, then reinstalled firmware through the webinstaller. Works fine as i see, flash works and intensity can be controlled with slider during creating reference image. Thanks.

**Comment by Jutika-ducky** [comment]
For everybody who encounter the same problem.

Quick fix without reassemble the board:
- finish the setup
- use the OTA Update ability and download the 15.7 "Update" Files from here: https://github.com/jomjol/AI-on-the-edge-device/releases/tag/v15.7.0

**Comment by pierafal** [comment]
might be related to: https://github.com/jomjol/AI-on-the-edge-device/issues/3471

**Comment by PsYcHLT** [comment]
Same here, following. Update to RC5 brings errors about camera or just stucks on delayed start.

**Comment by SybexX** [comment]
@PsYcHLT Did you only select the firmware.bin or the entire AI-on-the-edge-device__update__v16.0.0-RC5.zip when updating?
You can try the current main if you want: https://github.com/jomjol/AI-on-the-edge-device/actions/runs/12837723310/artifacts/2449558188
Don't forget to press CTRL+F5 after the update.

**Comment by PsYcHLT** [comment]
> -edge-device__update

I used full update zip file over OTA functionallity.
And this version should fix that?

So basically I just started using this app, and I love the idea and how it works so far! I have two ESP32 CAM devices, both were with 16.4RC version and both had same issue. One upgraded to 16.5RC - was not working properly, so downgraded to 15.7, everything seemed to be fine, but lately just wanted to give a one more try for 16.5 and now it don't even connect to the network :) Just to mention, always were using OTA functionality, I think I got to the point when I will have to dissasembly everything and flash it again to 15.7, or maybe version you have sent.

**Comment by SybexX** [comment]
@PsYcHLT A lot of things have been fixed in the main version.
If it doesn't connect, it could be that your SD is about to die, or something went wrong while copying the files.

**Comment by PsYcHLT** [comment]
@SybexX I'm getting this page in my browser, I think I'll also scan sd card for bads as I used some old scrap cards from back when 32GB sd card was really expensive 😆 

![Image](https://github.com/user-attachments/assets/75e8f0f4-0a8c-4a56-a202-65ffea6921f0)

**Comment by mesistostefre** [comment]
Same here. 15.7 works just fine for a long time. Tried the AI-on-the-edge-device__update__v16.0.0-RC5.zip to get the "measurement" state for raw in home assistant. It installed fine and worked first.
But once I adjusted the digits by 2 pixels, I got no flash but:
   AI on the Edge Device
   We have one or more critical errors:
   0x00000004
Hard to believe, the CAM is broken coincidently right after I installed the RC5...

**Comment by mesistostefre** [comment]
Ah ... just noticed ... the file system on the SDCard is bad. I checked it and the card itself is physically fine. It is the file system. So, maybe worth to check the source code for things like: file does not get closed properly once data has been written. Or a reboot gets executed while still writing a file, foo, bla...

**Comment by PsYcHLT** [comment]
@mesistostefre what filesystem you are using? NTFS? So it was OK for older versions but for newer it must be FAT32?

**Reply by caco3** [reply]
We always only supported FAT32!
See https://jomjol.github.io/AI-on-the-edge-device-docs/Installation/#notes and https://jomjol.github.io/AI-on-the-edge-device-docs/Frequent-Reboots/#sd-card-wrong-filesystem-only-fat32-is-supported

**Comment by mesistostefre** [comment]
FAT ... and now the whole thing (16 RC5) works for me.

**Comment by PsYcHLT** [comment]
Well, seems like my case is much weirder, no bads, card is FAT32. Will try @SybexX suggested version in a moment

**Comment by PsYcHLT** [comment]
Ended up fully reflashing it to 15.7 and then upgraded to last commit version of main branch. Seems to be working fine, also when taking reference image it turns on the LED now. WIll need more time to test it fully, but happy bunny for now 😸

**Comment by PsYcHLT** [comment]
Here we go again... So it was working fine, until I have painted my points of interest and restarted... So live video was working, I have aligned it, took a reference image (this time it turned flashlight), painted ROIs and boom...
<img width="1418" alt="image" src="https://github.com/user-attachments/assets/f886f32d-b3b0-4896-b5db-c3c3fb59960b" />

**Reply by PsYcHLT** [reply]
It will be not a discussion but my brain dump 😆 Changed power supply and it works again. Maybe that was the real issue after all

**Comment by mesistostefre** [comment]
Am using a calibrated external power supply, not one of these plugs. It certainly was not the power supply in my case.

**Comment by svenson08** [comment]
Same problem as @SGXander, only without an error message in the log. Only firmware 16.0.0-RC4 with web interface 15.7.0 solves the problem.

Unfortunately, the combination causes the problem that settings made via the web interface apparently write settings that are no longer valid in firmware 16 into the .ini. This can only be avoided by editing the ini file directly.
